projects/ngx-charts-on-fhir/src/lib/data-layer-list/annotation-options/annotation-options.component.spec.ts

```diff
@@ -0,0 +1,105 @@
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { ColorPickerModule } from '../../color-picker/color-picker.module';
+import { Component, forwardRef } from '@angular/core';
+
+@Component({
+  selector: 'color-picker',
+  template: '',
+  providers: [{ provide: NG_VALUE_ACCESSOR, multi: true, useExisting: forwardRef(() => MockColorPickerComponent) }],
+})
+class MockColorPickerComponent implements ControlValueAccessor {
+  writeValue(obj: any): void {
+    /* intentionally empty stub */
+  }
+  registerOnChange(fn: any): void {
+    /* intentionally empty stub */
+  }
+  registerOnTouched(fn: any): void {
+    /* intentionally empty stub */
+  }
+  setDisabledState?(isDisabled: boolean): void {
+    /* intentionally empty stub */
+  }
+}
+
+describe('AnnotationOptionsComponent', () => {
+  let component: AnnotationOptionsComponent;
+  let fixture: ComponentFixture<AnnotationOptionsComponent>;
+  let colorService: DataLayerColorService;
+  let palette: string[] = ['#FFFFFF', '#121212', '#000000'];
+  let loader: HarnessLoader;
+
+  beforeEach(async () => {
+    colorService = new DataLayerColorService(palette);
+    await TestBed.configureTestingModule({
+      imports: [ReactiveFormsModule, NoopAnimationsModule, MatInputModule, MatFormFieldModule],
+      declarations: [AnnotationOptionsComponent, MockColorPickerComponent],
+      providers: [FormBuilder, { provide: DataLayerColorService, useValue: colorService }],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(AnnotationOptionsComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+    loader = TestbedHarnessEnvironment.loader(fixture);
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should update annotation label when input is changed', async () => {
+    let emitted: any = null;
+    component.annotation = { label: { content: 'Old' } };
+    component.annotationChange.subscribe(e => emitted = e)
+    const labelInputHarness = await loader.getHarness(MatInputHarness.with({ selector: "[id='annotationLabel']" }));
+    await labelInputHarness.setValue('New');
+    expect(emitted.label.content).toEqual('New');
+  });
+
+  it('should update annotation yMax when input is changed', async () => {
+    let yMaxInputHarness = await loader.getHarness(MatInputHarness.with({ selector: "[id='yMax']" }));
+    let yMaxInputControl = fixture.componentInstance.form.controls['yMax'];
+    await yMaxInputHarness.setValue('100');
+    expect(yMaxInputControl.value).toBe(100);
+  });
+
+  it('should update annotation yMin when input is changed', async () => {
+    let yMinInputHarness = await loader.getHarness(MatInputHarness.with({ selector: "[id='yMin']" }));
+    let yMinInputControl = fixture.componentInstance.form.controls['yMin'];
+    await yMinInputHarness.setValue('100');
+    expect(yMinInputControl.value).toBe(100);
+  });
```

Comment:
these two tests still need to be updated to check the emitted event instead of FormControl value